server/utils/azureAppInsights.test.ts

```diff
@@ -0,0 +1,76 @@
+import { DataTelemetry, EnvelopeTelemetry } from 'applicationinsights/out/Declarations/Contracts'
+import { addUserDataToRequests, ContextObject } from './azureAppInsights'
+
+const user = {
+  activeCaseLoadId: 'LII',
+  isReceptionUser: true,
+  username: 'test-user',
+}
+
+const createEnvelope = (properties: Record<string, string | boolean>, baseType = 'RequestData') =>
+  ({
+    data: {
+      baseType,
+      baseData: { properties },
+    } as DataTelemetry,
+  } as EnvelopeTelemetry)
```

Comment:
oh nice! that's interesting that it can't cope with the partial object at the lower level

Reply:
Yeah thought it would have assumed it as well, so when intellij was telling me to define all the other fields thought that that was the standard process. Think your comment that if they are subset it should work - so thought maybe defining the subset would help it along the way 😁